lib/Sema/TypeCheckNameLookup.cpp

```diff
@@ -323,6 +325,14 @@ LookupResult TypeChecker::lookupUnqualified(DeclContext *dc, DeclName name,
       assert(foundInType && "bogus base declaration?");
     }
 
+    // This is a hack to filter out top level variables. If this flag is set
+    // then local variables are ignored, but for some reason, top level
+    // variables do not get ignored, so ignore them here.
```

Comment:
Top-level variables are member-qualified by the module, whereas local variables are unqualified.